mlir/tools/mlir-tblgen/OpInterfacesGen.cpp

```diff
@@ -619,10 +638,15 @@ bool InterfaceGenerator::emitInterfaceDecls() {
   llvm::sort(sortedDefs, [](const Record *lhs, const Record *rhs) {
     return lhs->getID() < rhs->getID();
   });
+  for (const Record *def : sortedDefs)
+    forwardDeclareInterface(Interface(def));
   for (const Record *def : sortedDefs)
     emitInterfaceDecl(Interface(def));
   for (const Record *def : sortedDefs)
     emitModelMethodsDef(Interface(def));
+  for (const Record *def : sortedDefs)
+    emitInterfaceTraitDecl(Interface(def));
```

Comment:
fyi: `Interface` ctor can be rather expensive (it has a vector of base interfaces that it populates + vector of methods).
As we re-create the objects *4* times here, does it make sense to refactor? (I failed to do it immediately so it'd probably go into a separate patch).

FWIW tablegen is part of the build system so if it is slow, the whole build is slow.